docs/source/app_development/publish.rst

```diff
@@ -1,72 +1,200 @@
 .. _develop-apps:publish-app:
 
-*****************************************
-Publish your app on the AiiDAlab registry
-*****************************************
+*************************
+Publish your AiiDAlab app
+*************************
 
-To make your app available to other AiiDAlab users, please register it on the `AiiDAlab registry`_.
+Here we explain how an app can be made available to AiiDAlab users by registering it on the `AiiDAlab registry`_.
 
-How to register
-===============
+.. panels::
+   :container: container-lg pb-3
+   :column: col-lg-12 p-2
 
-.. warning::
+   **Prepare the app for registration**
 
-    To be sure to have the latest information on the ``metadata.json`` JSON schema, see `the registry's repository README <https://github.com/aiidalab/aiidalab-registry/blob/master/README.md>`__.
+   .. important::
 
-In preparation of registering your AiiDAlab app in the `AiiDAlab registry`_, ensure you have a valid and updated ``metadata.json`` file in the root of your app's repository.
-To see a full list of valid keys, go to `this README section <https://github.com/aiidalab/aiidalab-registry/blob/master/README.md#valid-keys-for-metadatajson>`__.
+       If the app was created with the app cookiecutter it should already have all necessary information and this step can be skipped.
 
-You should fork the `AiiDAlab registry`_ repository (click `here to fork <https://github.com/aiidalab/aiidalab-registry/fork>`__ or click the "Fork" button in the upper right corner of the repository's webpage.
-To learn more about forking a GitHub repository, see their documentation `here <https://docs.github.com/en/github/getting-started-with-github/fork-a-repo>`__.
+   In preparation of registering your AiiDAlab app in the `AiiDAlab registry`_, ensure you have a valid and updated ``setup.cfg`` file in the root of your app's repository.
+   The ``setup.cfg`` file can alternatively also be placed within a hidden ``.aiidalab/`` directory in which case it will be parsed with precedence.
 
-In your forked repository, you can create a new git branch and update the ``apps.json`` file found in the root of the repository.
-The update consists of an addition to the JSON file, e.g.:
+   AiiDAlab parses the ``setup.cfg`` file for metadata and will recognize keys both from the standard `[metadata] <https://setuptools.pypa.io/en/latest/userguide/declarative_config.html#metadata>`__ section and a dedicated ``[aiidalab]`` section.
```

Comment:
As an additional comment (not necessary to fix now, more a "note to self"): this dual approach introduces complexity and can be confusing to new users. I don't have a great alternative suggestion at the moment, but if we can find a way to avoid it that could be useful

Reply:
Yes, the trade of is the potential duplication of information. If all information could be parsed from the standard `metadata` block, that would be ideal I think.

Reply:
@ltalirz I would suggest that we stick to the currently recommended approach and then have a separate discussion about this. This is indeed important to resolve and we currently need the `[aiidalab]` block only for the `title` and the `categories` fields. The first one I have already recommended for removal and the second one could potentially be parsed from the classifiers? Or maybe from keywords in the description?

Reply:
> @ltalirz I would suggest that we stick to the currently recommended approach and then have a separate discussion about this.

Sounds good. My first comment was trying to point out that from the text it was not obvious to me what the recommended approach is.
I assume it is to use the [metadata] when possible, and the [aiidalab] otherwise?
I will address this in https://github.com/aiidalab/aiidalab/pull/234 as well


About the long-term solution: one the one hand, following standards is good; on the other hand AiiDAlab applications simply are not the same as python packages.
I would therefore say that using the [metadata] as the only way of specifying app metadata is risky, even if one can make it work for some time - there can always be other metadata we may need to add in the future that have no correspondence in a python package.